Mixpanel/Mixpanel.m

```diff
@@ -1854,8 +1852,33 @@ - (void)executeCachedEventBindings {
 
 @end
 
-#pragma mark - People
+#pragma mark - WatchExtensions
+@implementation Mixpanel (WatchExtensions)
+
+/** Called on the delegate of the receiver. Will be called on startup if the incoming message caused the receiver to launch. */
+- (void)session:(WCSession *)session didReceiveMessage:(NSDictionary<NSString *, id> *)message {
+    if ([Mixpanel isValidWatchSessionMessage:message]) {
+        [[Mixpanel sharedInstance] track:message[@"event"] properties:message[@"properties"]];
+    }
+}
+
+/** Called on the delegate of the receiver when the sender sends a message that expects a reply. Will be called on startup if the incoming message caused the receiver to launch. */
+- (void)session:(WCSession *)session didReceiveMessage:(NSDictionary<NSString *, id> *)message replyHandler:(void(^)(NSDictionary<NSString *, id> *replyMessage))replyHandler {
+    if ([Mixpanel isValidWatchSessionMessage:message]) {
+        [[Mixpanel sharedInstance] track:message[@"event"] properties:message[@"properties"]];
+        replyHandler(@{ @"success": @YES });
+    } else {
+        replyHandler(@{ @"success": @NO, @"message": @"Message is not a mixpanel message" });
+    }
+}
+
++ (BOOL)isValidWatchSessionMessage:(NSDictionary<NSString *, id> *)message {
+    return [[message objectForKey:@"$mp_message_type"] boolValue];
```

Comment:
We just need to check if `$mp_message_type` is contained in the dictionary. `boolValue` will cause issues if we are passing a number, for example.